python/pyspark/pandas/generic.py

```diff
@@ -904,6 +904,8 @@ def to_json(
 
         .. note:: output JSON format is different from pandas'. It always use `orient='records'`
             for its output. This behaviour might have to change in the near future.
+        
+        .. note:: If column have only NaN or Null values. The column well be deleted.
```

Comment:
There's `ignoreNullFields` option to control this behaviour. If pandas writes both in JSON, we will have to set this as `False` by default, and document this.

Reply:
So if I understand your correctly, it means either adding a keyword only argument

```python
    def to_json(
        self,
        ...,
        *,
        ignoreNullFields: bool = False,
        **options: Any,
    ) -> Optional[str]:
       ...
       options["ignoreNullFields"] = ignoreNullFields
```

or patching the `options`

```python
    def to_json(
        self,
        ...,
        **options: Any,
    ) -> Optional[str]:
       ...
       if "ignoreNullFields" not in options:
           options["ignoreNullFields"] = False

```
to ensure that output is consistent independent of `path`.

Reply:
Yes, this well be the best option. But are you thinking about changing this in https://github.com/apache/spark/pull/26098

// Whether to ignore column of all null during json generating
  val structIngoreNull = parameters.getOrElse("structIngoreNull", "true").toBoolean

or just change the “koalas” code? 

Yes, pandas writes both in JSON. I have updated the jira post with it.

Reply:
Let's change koalas code alone for now to match with its default behaviour same as pandas's.

This pandas API on Spark (or legacy Koalas) is pretty new in Spark (it's sort of "alpha component", https://spark.apache.org/versioning-policy.html) vs. other codes in PySpark are pretty old, and many users are dependent on its existing behavior. Also, pandas API on Spark tires to match its behaviour to pandas' whereas Spark does not necessarily follow.

Reply:
We could add a note like you can omit NaN or Null in JSON output by `ignoreNullFields` set to `True` which is a default behaivour of PySpark's JSON writer.

Reply:
> or just change the “koalas” code?

I concur with @HyukjinKwon ‒ only `pyspark.pandas` code should be changed, and that for the sake of the API consistence.

As of the main writer behavior, I am strongly against changing defaults at all (not only in this PR). In general, omitting NULL fields in the output is the preferred behavior and there is no good justification for flipping the defaults now, after API being stable for years.